plugins/hls-alternate-number-format-plugin/src/Ide/Plugin/Conversion.hs

```diff
@@ -0,0 +1,170 @@
+import           Numeric
+import           Text.Regex.TDFA               ((=~))
+
+data FormatType = IntFormat IntFormatType
+                | FracFormat FracFormatType
+                | AnyFormat AnyFormatType
+                | NoFormat
+                deriving (Show, Eq, Generic)
+
+instance NFData FormatType
+
+data IntFormatType = HexFormat
+                   | OctalFormat
+                   | BinaryFormat
+                   | NumDecimalFormat
+                   deriving (Show, Eq, Generic)
+
+instance NFData IntFormatType
+
+data FracFormatType = HexFloatFormat
+                    | ExponentFormat
+                    deriving (Show, Eq, Generic)
+
+instance NFData FracFormatType
+
+data AnyFormatType = DecimalFormat
+                   deriving (Show, Eq, Generic)
+
+instance NFData AnyFormatType
+
+alternateFormat :: [FormatType] -> Literal -> [Text]
+alternateFormat fmts lit = case lit of
+  IntLiteral _ (Just _) val  -> concatMap (alternateIntFormat val) (removeCurrentFormat lit fmts)
+  FracLiteral _ (Just _) val -> if denominator val == 1 -- floats that can be integers we can represent as ints
+      then concatMap (alternateIntFormat (numerator val)) (removeCurrentFormat lit fmts)
+      else concatMap (alternateFracFormat val) (removeCurrentFormat lit fmts)
+  _                          -> ["Uh Oh"] -- This means there is no Source Text so we just ignore it
+
+alternateIntFormat :: Integer -> FormatType -> [Text]
+alternateIntFormat val fmt = case fmt of
+  IntFormat ift           -> case ift of
+    HexFormat        -> [T.pack $ toHex val]
+    OctalFormat      -> [T.pack $ toOctal val]
+    BinaryFormat     -> [T.pack $ toBinary val]
+    NumDecimalFormat -> toNumDecimal val  -- this is the only reason we return List of Text :/
+  AnyFormat DecimalFormat -> [T.pack $ toDecimal val]
+  _                       -> []
+
+alternateFracFormat :: Rational -> FormatType -> [Text]
+alternateFracFormat val fmt = case fmt of
+  AnyFormat DecimalFormat   -> [T.pack $ toFloatDecimal (fromRational val)]
+  FracFormat ExponentFormat -> [T.pack $ toFloatExpDecimal (fromRational val)]
+  FracFormat HexFloatFormat -> [T.pack $ toHexFloat (fromRational val)]
+  _                         -> []
+
+removeCurrentFormat :: Literal -> [FormatType] -> [FormatType]
+removeCurrentFormat lit fmts = case getSrcText lit of
+    Just src -> foldl (flip delete) fmts (sourceToFormatType src)
+    Nothing  -> fmts
+
+hexRegex, hexFloatRegex, binaryRegex, octalRegex, decimalRegex, numDecimalRegex :: Text
+hexRegex = "0[xX][a-fA-F0-9]+"
+hexFloatRegex = "0[xX][a-fA-F0-9]+\\.[a-fA-F0-9]+(p[+-]?[0-9]+)?"
+binaryRegex = "0[bB][0|1]+"
+octalRegex = "0[oO][0-8]+"
+decimalRegex = "\\d+"
+numDecimalRegex = "[0-9]+\\.[0-9]+[eE][+-]?[0-9]+"
+
+-- we export the regex for tests, but we want to be explicit in our matches
+-- so we need to match the beginning/end of the source text
+wrap :: Text -> Text
+wrap regex = "^" <> regex <> "$"
```

Comment:
If this is exported, you should probably document it via a haddock comment so other users have an easier time figuring out what this does.

Also, considering this function is going to be exported, `wrap` does not really help with understanding what this does.
Maybe `matchLineRegex` or something like that expresses the intent better?